File: src/IdenticalArrays.java
//{ Driver Code Starts
import java.util.*;


// } Driver Code Ends
// User function Template for Java
class Solution2 {
    public boolean isIdentical(List<Integer> arr1, List<Integer> arr2) {
        for ( Integer elem : arr1 ){
            if ( !arr2.contains(elem) || Collections.frequency(arr1, elem) != Collections.frequency(arr2, elem) ) {
                return false;
            }
        }
        return true;
    }
}

//{ Driver Code Starts.
public class IdenticalArrays {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine());
        while (t-- > 0) {
            String line = sc.nextLine();
            List<Integer> arr1 = new ArrayList<>();
            Scanner lineScanner = new Scanner(line);
            while (lineScanner.hasNextInt()) {
                arr1.add(lineScanner.nextInt());
            }
            line = sc.nextLine();
            List<Integer> arr2 = new ArrayList<>();
            lineScanner = new Scanner(line);
            while (lineScanner.hasNextInt()) {
                arr2.add(lineScanner.nextInt());
            }
            Solution2 ob = new Solution2();
            boolean ans = ob.isIdentical(arr1, arr2);
            if (ans) {
                System.out.println("true");
            } else {
                System.out.println("false");
            }
            lineScanner.close();

            System.out.println("~");
        }
        sc.close();
    }
}
// } Driver Code Ends

File: cpp/identical_arrays.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    bool isIdentical(vector<int> &arr1, vector<int> &arr2)
    {
        // directly checking the size
        if (arr1.size() != arr2.size())
            return false;

        // checking the count of each element
        std::unordered_map<int, int> map;

        // creating ref table;
        for (auto elem : arr1)
        {
            map[elem]++;
        }

        for (auto elem : arr2)
        {
            map[elem]--;
            if (map[elem] < 0)
            {
                return false;
            }
        }

        // if all tests passed
        return true;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cout << "Number of cases: " , cin >> t;
    cin.ignore();
    while (t--) {
        string s;
        cout << "arr1: ";
        getline(cin, s);
        stringstream ss(s);
        vector<int> arr1, arr2;
        int num;
        while (ss >> num) {
            arr1.push_back(num);
        }
        cout << "arr2: ";
        getline(cin, s);
        ss.clear();
        ss.str(s);
        while (ss >> num) {
            arr2.push_back(num);
        }
        Solution ob;
        bool ans = ob.isIdentical(arr1, arr2);
        if (ans) {
            cout << "true" << endl;
        } else {
            cout << "false" << endl;
        }
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends